miner/payload_building.go

```diff
@@ -95,7 +114,7 @@ func (payload *Payload) Resolve() *beacon.ExecutableDataV1 {
 	select {
 	case <-payload.stop:
 	default:
-		close(payload.stop)
+		//	close(payload.stop)
```

Comment:
Why do you not stop payload building on resolve?
I think the correct way according to spec is to return the payload once we get a `GetPayload` call and return the *same* payload on all subsequent calls

Reply:
Right, it's a very good point actually. It might lead to some troubles that validator produces different blocks in a single slot?

Reply:
Oh, really? So `GetPayload` needs to terminate the building. I can change it so it works like that, but could you elaborate on the downsides of _not_ doing that? Just so I have the full picture.

Reply:
It makes the call non-deterministic and a bit racy. Suppose you call `GetPayload` twice by accident and get two different results. Now you need to decide which of those two completely valid payloads you want to use

Reply:
Hmm I think it is kind of an arbitrary decision, but its part of the spec afair

Reply:
> It makes the call non-deterministic and a bit racy. Suppose you call GetPayload twice by accident and get two different results

That's already in the design, isn't it? You first get an empty block, then you get a non-empty. 

> arbitrary decision, but its part of the spec afair

Ok

Reply:
While, it's different. Recommit just updates the payload internally, all generated payloads are not available from the CL side. But if we don't stop building after retrieval, then different versions payload may all be accessible for CL.